### DISCUSSION TITLE
20231005 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by cgbosse** [nautilus-cyberneering/.github#61] [Meetings]
Date: 2023-10-05

Time: 8:30 UTC (10:30 in Vaduz)

Location:
<https://meeting.hodl10.io/ConsistentClassroomsMergeSideways>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- CGarnham(@da2ce7)  - 
- JCelano(@josecelano) - 
- CBosse(@cgbosse) - 
- MMelchor(@mario-nt) - 
- GByrne(@grmbyrn) - 

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by mario-nt** [comment]
**GUI**
- Worked on https://github.com/torrust/torrust-index-gui/issues/277

**INDEX**

- Worked on https://github.com/torrust/torrust-index/issues/303
- Worked on https://github.com/torrust/torrust-index/issues/39
- Worked on https://github.com/torrust/torrust-index/discussions/312
- Read https://github.com/torrust/torrust-index/issues/311
- Worked on https://github.com/torrust/torrust-website/issues/81

**Other**
- Worked on https://github.com/torrust/torrust-index-api-lib/issues/37
- Worked on https://github.com/torrust/torrust-index-types-lib/issues/12


**Training**

- Research on Continuous integration
- Docker 101
- Learnt about Rust, clone() vs copy errors
- Reviewing CI errors

**Comment by grmbyrn** [comment]
## Secure Git Guide
* Added home page to project
* Implemented articles into home page so that each link routes to correct article
* Split project into appropriate components
* Styled project CSS to improve user experience

## TypeScript
* Completed TypeScript course - https://scrimba.com/learn/typescript

## AI
* Bought $5 credit to use OpenAI API
* Began course on AI, using OpenAI API to create a movie pitchbot project https://scrimba.com/learn/buildaiapps
* Pushed code for AI project to GitHub https://github.com/grmbyrn/pitchbot-ai

## Medium
* Wrote articles on TypeScript - https://medium.com/@grmbrn89

**Comment by josecelano** [comment]
### Torrust Index

I was searching for alternatives to `serde-bencode` because of this issue: 

- https://github.com/torrust/torrust-index/issues/266

The research:

- https://github.com/torrust/torrust-index/issues/311

After the research, I decided to fork the `serde-bencode` repo to fix the problem, but while I was working on the fork, I realized the problem was already fixed, but a new version had not been published yet.

To fix the problem, you need to:

- Use the latest commit from the repo.
- Avoid using tuple structs like the one we are using:

```rust
pub nodes: Option<Vec<TorrentNode>>,

pub struct TorrentNode(String, i64);
```

Instead, we have to use:

```rust
pub nodes: Option<Vec<(String, i64)>>,
```

The data structure we are trying to deserialize:

```json
{
   "info": {
      "length": 8,
      "name": "minimal.txt",
      "piece length": 16384,
      "pieces": "<hex>30 9A 84 51 4A F1 09 D0 8C 34 42 11 26 67 7F 84 B3 23 4D 78</hex>"
   },
   "nodes": [
      [
         "188.163.121.224",
         56711
      ],
      [
         "162.250.131.26",
         13386
      ]
   ]
}
```

`nodes` is a bencoded list of lists.

### Torrust Serde Bencode

I've been working on a [forked](https://github.com/torrust/torrust-serde-bencode) version of the `serde-bencode` crate, but in the meantime, I've contacted the repo owner, and he gave me access to the project, so now I'm able to update the repo and publish a new minor version `v0.2.4`:

- https://crates.io/crates/serde_bencode

Tasks:

- Fix build and clipply errors/warnings.
- Update dependencies.
- Add workflows: formatting, testing, coverage, becnhmarking.
- Merge pending to merge PRs.
- Add unit tests. WIP.

### Cargo Pretty Test

I've also spent some hours on: https://github.com/josecelano/cargo-pretty-test

**Reply by josecelano** [reply]
I've opened a [draft PR](https://github.com/toby/serde-bencode/pull/34) to merge back in the upstream repo the changes I made in the fork.
